jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -65,6 +66,8 @@ def process(self, input):
             # Get the MSA metadata file if needed and add to reffiles
             msa_metadata_file = input_model.meta.instrument.msa_metadata_file
             if msa_metadata_file is not None:
+                if isinstance(input, str):
```

Comment:
Second @jdavies-st  in that, though not apparently stated anywhere, it is heavily implied that `process` should not be called directly. The real practical matter is that many other `process` definitions heavily rely on all sorts of parameters and setups. If one starts down the path of re-implementing setup steps when one is explicitly bypassing the setup, I fear for the future.

That being said, the `Step` API should allow one to easily do the step setup and `process` explicitly. At the moment, this is not the case: `Step.call` is actually close, but `Step.run` definitely has too much logic baked-in and could stand for a refactor.

So, for a simple issue as this with current architecture, the change is fine.

Reply:
Given that #2157 now implements a setter for step.input_dir, I think this is all I need.